CODEOWNERS

```diff
@@ -21,6 +21,8 @@ lte/gateway/release @kkahrs
 lte/gateway/python/magma/pipelined @koolzz @pshelar
 lte/gateway/python/integ_tests @ssanadhya @ulaskozat
 
+lte/cloud/go/protos @magma/entire-org
```

Comment:
@themarwhal correct me if I'm wrong but I think we actually want this to be `protos/*.pb.go`

Reply:
Thanks for the feedback. Are we wanting to match ALL `*.pb.go` files in any directory or just those under `lte/cloud/go/protos`?

CODEOWNERS follows the gitignore pattern syntax[1]. So an entry like 

> `lte/cloud/go/protos/*.pb.go`

would only apply to files in that particular directory. I believe a double `**` would match recursively. Something like 

> `lte/cloud/go/protos/**/*.pb.go`


1. https://git-scm.com/docs/gitignore#_pattern_format

Reply:
I see. Yeah my assumption then is we want something like below, to target all generated Go code. Perhaps @ssanadhya can confirm

```
*/protos/*.pb.go
*/models/*_swaggergen.go
```

Reply:
Ideally we can do a pattern

```
*.pb.go
*_swaggergen.go
```

Which matches across the whole repo.

Reply:
Yes @hcgatewood , my thinking was for the generated files

Reply:
Agreed with @hcgatewood above, i.e. we want to target all generated Go code, under orc8r as well as gateways.
@xjtian, it will be better to have some directory paths specified to ensure generated code stays in specific locations. Adding a blanket rule for all files with matching patterns can bleed generated code.

Reply:
Considering the multiple levels of nesting in different directories, I think we need to tweak the pattern from @hcgatewood:

```
**/protos/*.pb.go
**/models/*_swaggergen.go
```

This would match MOST of the existing files that would also be picked up by @xjtian's pattern of all pb.go and _swaggergen.go files, except existing ones in the following directories:

```
/cwf/cloud/go/protos/mconfig/
/devmand/cloud/go/protos/mconfig/
/feg/cloud/go/tools/migrations/m003_configurator/plugin/types/
/lte/cloud/go/tools/migrations/m003_configurator/plugin/types/
/orc8r/cloud/go/services/configurator/storage/
/orc8r/cloud/go/services/streamer/test_utils/mconfig/test_protos/
/orc8r/cloud/go/tools/migrations/m001_config_service/migration/
/orc8r/cloud/go/tools/migrations/m002_cleanup_legacy_configs/migration/
/orc8r/cloud/go/tools/migrations/m003_configurator/plugin/types/
/orc8r/gateway/go/mconfig/test/testcfg/
/wifi/cloud/go/protos/mconfig/
/wifi/cloud/go/tools/migrations/m003_configurator/plugin/types/
```

If this is the desire state, I can push a new commit with the pattern from above or at `types` and `mconfig` as additional directories to match.

Reply:
Would something like `**/*.pb.go`, `**/*_swaggergen.go` work? Files with these suffixes are always generated files.

Reply:
That and the previous pattern you suggested would both match the entire repo, but it sounded like @ssanadhya was concerned about allowing sprawl where those files might end up anywhere. By doing something that's a true directory wildcard and enabling an entry in CODEOWNERS for it, I believe the result would be that anyone could push a file matching that pattern anywhere in the repository. I'm actually unclear if that's the desired result or not.

Reply:
I think that's fine for these generated files in the Go codebase actually. We used to have this duct taped build system that relied on specific directory names for generated files but that hasn't been the case for a while now so it's totally fine to stick proto/swagger files in whatever packages.

Reply:
@jbryce Should we still try to merge this? Or have we decided against it?

Reply:
I'll push a new commit with the directory wildcard

Reply:
New commit is up. This will allow anyone in the entire-org group to merge files anywhere in the repo matching this pattern, including generated code or any file created that ends in .pb.go or _swaggergen.go.